src/tabnineChatWidget/handlers/context/basicContextHandler.ts

```diff
@@ -36,7 +36,76 @@ async function noEditorResponse() {
       );
     }
   }
+
   return {
     metadata,
+    language: await getPredominantWorkspaceLanguage(),
   };
 }
+
+async function getPredominantWorkspaceLanguage(
+  maxFiles: number = 50
+): Promise<string | undefined> {
+  const breakdown: { [extension: string]: number } = {};
+
+  // Getting the known languages from VS Code
+  const knownLanguages = await vscode.languages.getLanguages();
+
+  // File extension to VS Code language ID mapping
+  const fileTypeToLanguageID = new Map<string, string>([
+    ["js", "javascript"],
+    ["py", "python"],
+    ["java", "java"],
+    ["cpp", "cpp"],
+    ["hpp", "cpp"],
+    ["c", "c"],
+    ["h", "c"],
+    ["cs", "csharp"],
+    ["php", "php"],
+    ["ts", "typescript"],
+    ["rb", "ruby"],
+    ["go", "go"],
+    ["rs", "rust"],
+    ["swift", "swift"],
+    ["kt", "kotlin"],
+    ["dart", "dart"],
+    ["json", "json"],
+    ["yaml", "yaml"],
+    ["yml", "yaml"],
+  ]);
+
+  const topFileTypes = Array.from(fileTypeToLanguageID.keys());
+
+  // Ensure a workspace is open
+  if (!vscode.workspace.workspaceFolders) {
+    return;
+  }
+
+  // Get up to maxFiles from the workspace without depth restriction
+  const files = await vscode.workspace.findFiles("**/*", null, maxFiles);
```

Comment:
few notes, 
1 - we already have a lang to ext mapping [here](https://github.com/codota/tabnine-vscode/blob/cc74eeec67703c2165071d51176cfaa1a98b7c55/src/globals/languages.ts#L1) 
2- you cab use workspace.textDocuments in order to get the correct languageId without the ext<>language mapping.

then the code could be something like:
```
function getMostUsedLanguage() {
  const languageCounts: { [key: string]: number } = {};

  workspace.textDocuments.forEach((document) => {
    const language = document.languageId;

    if (languageCounts[language]) {
      languageCounts[language] += 1;
    } else {
      languageCounts[language] = 1;
    }
  });

  const mostUsedLanguage = Object.keys(languageCounts).reduce((a, b) =>
    languageCounts[a] > languageCounts[b] ? a : b
  );
  return mostUsedLanguage;
}

Reply:
+ add tests for it

Reply:
Tests added, the workspace.textDocuments doesn't work when all files are closed.